dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -350,16 +361,23 @@ private Object parseFlatField(FieldDef fd, boolean isLittleEndian) throws Unpars
                 if (fd.isUnsigned())
                 {
                     intValue = getUnsigned((int) fd.getDataLen(), fd.getSourceType() == HpccSrcType.LITTLE_ENDIAN);
-                    if (intValue < 0)
+                    fieldValue = Long.valueOf(intValue);
+
+                    if (useDecimalForUnsigned8 && fd.getDataLen() == 8)
+                    {
+                        BigInteger bi = extractUnsigned8Val(intValue);
+                        fieldValue = new BigDecimal(bi);
+                    }
+                    else if (intValue < 0)
                     {
                         messages.addMessage("Warning: Possible unsigned overflow in column: '" + fd.getFieldName() + "'. Convert values to BigInteger via org.hpccsystems.commons.utils.extractUnsigned8 if necessary." );
                     }
                 }
                 else
                 {
                     intValue = getInt((int) fd.getDataLen(), fd.getSourceType() == HpccSrcType.LITTLE_ENDIAN, fd.isBiased());
+                    fieldValue = Long.valueOf(intValue);
                 }
-                fieldValue = Long.valueOf(intValue);
```

Comment:
In ThorFileReader you have

                       if (dataLen == 4 && !isUnsigned) {
                            fieldValue = Integer.valueOf((int) intValue);
                        } else {
                            fieldValue = Long.valueOf(intValue);
                        }

is that check needed here?

Reply:
@drealeed good question, that check is Tardis specific to match the behavior in the Tardis IRecordWriter. So, we don't need it here.